src/core/server/rendering/rendering_provider.tsx

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { renderToStaticMarkup } from 'react-dom/server';
+import { format } from 'prettier';
```

Comment:
Prettier serves a couple roles here:

- Makes it so the indentation is consistent from the string-building of all the template strings across multiple components
- Ensures the formatting is consistent for snapshot tests

The formatted content should be negligible in size. We should consider using an HTML minifier in the future to negate this need.

Reply:
Prettier serves a couple roles here:

- Makes it so the indentation is consistent from the string-building of all the template strings across multiple components
- Ensures the formatting is consistent for snapshot tests

The formatted content should be negligible in size. We should consider using an HTML minifier in the future to negate this need.

Reply:
what overhead it adds in runtime? 

> Ensures the formatting is consistent for snapshot tests

we can use prettier in tests only.
```
expect(format(render(...))).toMatchSnapshot();
```

Reply:
> what overhead it adds in runtime?

I haven't done any benchmarking, but I can't imagine it would be very negligible given we have control over the content and it is relatively small. For text of this size, parsing should be fast as well, since it is a single file and doesn't need any external static analysis. I do feel like the slight performance tradeoff is worth having consistent and predictable output.

> what if prettier cannot parse HTML ? users won't get the page

Prettier has several parsers, including HTML. In fact, we use the HTML parser in the implementation:

```ts
return format(markup, { parser: 'html' });
```

Reply:
My point was that all parsers have bugs and HTML is not a  trivial language to parse. Prettier has 37 open issues for the HTML parser. https://github.com/prettier/prettier/issues?q=is%3Aopen+is%3Aissue+label%3Alang%3Ahtml
It means that a user cannot receive a page whenever prettier tripped us up.